bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/service/BusinessPartnerService.kt

```diff
@@ -209,6 +219,26 @@ class BusinessPartnerService(
         }
 
 
+    private fun requestNewCleaning(businessPartner: BusinessPartner): TaskCreateResponse {
+        return orchestrationApiClient.goldenRecordTasks.createTasks(
+            TaskCreateRequest(
+                TaskMode.UpdateFromSharingMember,
+                listOf(businessPartnerMappings.toBusinessPartnerGenericDto(businessPartner))
+            )
+        )
+    }
+
+    private fun setUpNewSharingState(taskCreateResponse: TaskCreateResponse, sharingStateType: SharingStateType, externalId: String) {
```

Comment:
I would maybe change it to `updateSharingState(entity: BusinessPartner, stateDto: TaskClientStateDto)`.
This way it can be reused later when searching for task state updates.
A `sharingStateType` parameter is not really needed because there is a matching `resultState` in `TaskProcessingStateDto`.